src/toffy/fov_watcher.py

```diff
@@ -208,21 +213,14 @@ def __init__(
             for name in files:
                 self.on_created(FileCreatedEvent(os.path.join(root, name)))
 
-    def on_created(self, event: FileCreatedEvent):
-        """Handles file creation events
-
-        If FOV structure is completed, the fov callback, `self.fov_func` will be run over the data.
-        This function is automatically called; users generally shouldn't call this function
-
-        Args:
-            event (FileCreatedEvent):
-                file creation event
-        """
-        super().on_created(event)
-
+    def _run_callbacks(self, event: Union[DirCreatedEvent, FileCreatedEvent, FileMovedEvent]):
         # check if what's created is in the run structure
         try:
-            fov_ready, point_name = self.run_structure.check_run_condition(event.src_path)
+            if type(event) in [DirCreatedEvent, FileCreatedEvent]:
+                file_trigger = event.src_path
+            else:
+                file_trigger = event.dest_path
```

Comment:
So if the original file is called `fov-n-scan-1.bin.asdf`, the destination path would then be `fov-n-scan-1.bin`?

Reply:
Yes